zebra-rpc/src/methods.rs

```diff
@@ -248,6 +256,34 @@ where
         }
         .boxed()
     }
+
+    fn get_best_block_hash(&self) -> BoxFuture<Result<GetBestBlockHash>> {
+        let mut state = self.state.clone();
+
+        async move {
+            let request = zebra_state::Request::Tip;
+            let response = state
+                .ready()
+                .and_then(|service| service.call(request))
+                .await
+                .map_err(|error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                    data: None,
+                })?;
+
+            match response {
+                zebra_state::Response::Tip(Some((_height, hash))) => Ok(GetBestBlockHash(hash)),
+                zebra_state::Response::Block(None) => Err(Error {
+                    code: ErrorCode::ServerError(0),
+                    message: "Block not found".to_string(),
+                    data: None,
+                }),
```

Comment:
This will panic if the state is empty:
```suggestion
                zebra_state::Response::Tip(None) => Err(Error {
                    code: ErrorCode::ServerError(0),
                    message: "No blocks in state".to_string(),
                    data: None,
                }),
```

Reply:
Sorry about that, fixed in https://github.com/ZcashFoundation/zebra/pull/3754/commits/ddfabce08f1dc9dc00398b2143ad17e2d1aa0acb